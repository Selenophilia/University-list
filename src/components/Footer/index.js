import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Button, TextField, Typography } from "@material-ui/core";
import CopyrightIcon from "@material-ui/icons/Copyright";
const useStyles = makeStyles(() => ({
  footer: {
    height: 350,
    backgroundColor: "#6596F3",
    padding: 100
  },
  title: {
    color: "#fff",
    fontSize: 32,
    fontWeight: "bold"
  },
  input: {
    marginTop: 20,
    backgroundColor: "#fff",
    borderRadius: 8,
    width: 300,
    border: "none",
    "&::before": {
      borderBottom: "none"
    },
    "&::after": {
      borderBottom: "none"
    }
  },
  submit: {
    backgroundColor: "#3CB371",
    color: "#FFFFFF",
    width: 90,
    height: 57,
    borderTopLeftRadius: 0,
    borderBottomLeftRadius: 0,
    position: "relative",
    top: 20,
    left: -90,
    "&:hover": {
      backgroundColor: "#38a167"
    },
    boxShadow: "none"
  },
  line: {
    marginTop: 30
  },
  copyright: {
    marginLeft: 10
  },
  footerText: {
    color: "#ffff",
    display: "flex",
    width: 400
  }
}));
const Footer = () => {
  const classes = useStyles();
  return (
    <div className={classes.footer}>
      <footer>
        <Typography className={classes.title} variant="p">
          Subscribe to our news letter
        </Typography>
        <div className={classes.subsribe}>
          <TextField
            id="outlined-basic"
            label="Suscribe"
            variant="filled"
            className={classes.input}
          />
          <Button variant="contained" className={classes.submit}>
            {"Submit"}
          </Button>
        </div>
        <hr className={classes.line} />
        <div className={classes.footerText}>
          <CopyrightIcon />
          <Typography className={classes.copyright} variant="p">
            copyright university listings 2021
          </Typography>
        </div>
      </footer>
    </div>
  );
};
export default Footer;
