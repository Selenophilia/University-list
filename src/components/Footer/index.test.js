import React from "react";
import { render } from "@testing-library/react";
import { ApolloProvider } from "@apollo/client";
import client from "../../utils/client";
import Footer from ".";

const renderRoot = () => {
  return render(
    <ApolloProvider client={client}>
      <Footer />
    </ApolloProvider>
  );
};

it("renders Footer component correctly ", () => {
  const { asFragment } = renderRoot();
  expect(asFragment()).toMatchSnapshot();
});
