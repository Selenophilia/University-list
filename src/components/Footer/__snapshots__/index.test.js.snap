// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer component correctly  1`] = `
<DocumentFragment>
  <div
    class="makeStyles-footer-1"
  >
    <footer
      class="makeStyles-footerContent-2"
    >
      <span
        class="MuiTypography-root makeStyles-title-3"
      >
        Subscribe to our news letter
      </span>
      <div>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-input-5"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
            data-shrink="false"
            for="outlined-basic"
            id="outlined-basic-label"
          >
            Suscribe
          </label>
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input"
              id="outlined-basic"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-6"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Submit
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <hr
        class="makeStyles-line-7"
      />
      <div
        class="makeStyles-footerText-9"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.88 9.14c1.28.06 1.61 1.15 1.63 1.66h1.79c-.08-1.98-1.49-3.19-3.45-3.19C9.64 7.61 8 9 8 12.14c0 1.94.93 4.24 3.84 4.24 2.22 0 3.41-1.65 3.44-2.95h-1.79c-.03.59-.45 1.38-1.63 1.44-1.31-.04-1.86-1.06-1.86-2.73 0-2.89 1.28-2.98 1.88-3zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
          />
        </svg>
        <span
          class="MuiTypography-root makeStyles-copyright-8"
        >
          copyright university listings 2021
        </span>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
